cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -154,8 +154,11 @@ object PaybackFlow {
                     val cashState = serviceHub.toStateAndRef<CashState>(stx.tx.inputs.last()).state.data
 
                     "Payback value differ from borrowed amount." using (cashState.value.compareTo(iouState.value) == 0)
-                    "Lender should own IOU." using (iouState.lender.owningKey == ourIdentity.owningKey)
-                    "Borrower should own payback cash." using (iouState.borrower.owningKey == cashState.owner.owningKey)
+                    "Lender should own IOU." using iouState.lender.owningKey.equals(ourIdentity.owningKey)
+                    "Borrower should own payback cash." using iouState.borrower.owningKey.equals(cashState.owner.owningKey)
+                    // below should be always true, because on cash creation we have condition
```

Comment:
Not always true because another flow could have created Monopoly money. So your check is valid.